### PR TITLE
feat: add `cargo run` runner

### PR DESCRIPTION
**Pull request by doinkythederp**
## Describe the changes this PR makes. Why should it be merged?

This PR adds a cargo runner config, allowing contributors to succinctly build, upload, start, and view the serial output of examples by simply typing `cargo run --example <EXAMPLE>`.

It also works with Rust Analyzer:

image

## Additional Context
- I have tested these changes on a VEX V5 brain.
- I have tested these changes on Mac, Linux (nix flake on Debian) and Windows.

### CODE REVIEW

**Comment by doinkythederp** [comment]
I might refactor this to live in cargo-pros so I don't need to add a dependency.

**Comment by doinkythederp** [comment]
Requires a recent git build of cargo-pros to work properly.

**Comment by doinkythederp** [comment]
Flake lock needs to be updated before a merge

**Review by Gavin-Niederman — CHANGES_REQUESTED** [review]
I dislike the fact that you can only upload to slot one. There might be some way around this, but it feels wrong to me to do this.

File: README.md [hunk 1]
```diff
@@ -10,13 +10,17 @@ vexide is still in development but we are quickly moving towards competition rea
 
 ## Usage
 
+After setting up your development environment, connect to a VEX V5 via USB and run `cargo run --example basic` to execute the basic example on the V5 brain. Try the other examples like `screen` or `clawbot`, or create your own vexide project from the [template](https://github.com/vexide/vexide-template).
```
Comment (Gavin-Niederman) [inline README.md:13]: You should clarify that cargo run doesn't work in the template and that you have to use cargo pros.

**Review by Gavin-Niederman — CHANGES_REQUESTED** [review]

**Comment by Tropix126** [comment]
Given the recent release of cargo-v5, i'm thinking we might want to write a new runner that's also present in the template for this. (just runs `cargo v5 upload --after=run && cargo v5 terminal`). Slot number limitations are no longer a concern since those are either prompted or configured in Cargo.toml now.

I'm not sure if the runner functionality should be all decentralized as it might be worthwhile to push updates to it though, so maybe we should slap it into a crate or integrate it into `cargo v5`.

**Comment by doinkythederp** [comment]
> Given the recent release of cargo-v5, i'm thinking we might want to write a new runner that's also present in the template for this. (just runs `cargo v5 upload --after=run && cargo v5 terminal`). Slot number limitations are no longer a concern since those are either prompted or configured in Cargo.toml now.
> 
> I'm not sure if the runner functionality should be all decentralized as it might be worthwhile to push updates to it though, so maybe we should slap it into a crate or integrate it into `cargo v5`.

The existence of `cargo pros run` for compiling and uploading and running and terminal at the same time was actually necessary because otherwise there is a risk of serial data being lost between the two shell commands as the program runs. Also the naive approach of chaining the two commands together presumably wouldn't work on windows because of the &&. Thus I think it should stay integrated into the cli.

**Comment by doinkythederp** [comment]
I get back from traveling tomorrow so I might be able to work on implementing `cargo v5 run` so this PR can be updated.

**Comment by Tropix126** [comment]
Sounds good. The terminal issue can now be circumvented as we have the ability to start/stop programs at any time.

**Comment by doinkythederp** [comment]
Ready for review

**Comment by doinkythederp** [comment]
This would be pretty useful to add to the template too to increase usability for people who aren't familiar with the terminal run button

**Review by Gavin-Niederman — APPROVED** [review]
LLLGMGGGMMMTMMMMMMM 
{2B6D9742-0EAA-4FCD-98C7-84DDC3E7861D}

**Review by Gavin-Niederman — APPROVED** [review]
LGMTLMTLMMMLLMGMMMMMMMMMMMMMMM
{09924143-944A-441D-9379-C293AD6C0DB4}
M :4 :3 :3

**Pull request merged**